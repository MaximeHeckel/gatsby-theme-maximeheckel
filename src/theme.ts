const theme = {
  light: {
    backgroundColor: 'var(--maximeheckel-colors-body-0, #F6F7F8)',
    borderColor: '#f5f5f9',
    boxShadow: '0 0px 12px -6px rgba(0, 24, 40, 0.25)',
    colors: {
      blue: 'var(--maximeheckel-colors-brand, #5184f9)',
      gray: '#F8F8F9',
      pink: '#FED5D7',
      prism: {
        token: `#fff`,
        // tslint:disable-next-line:object-literal-sort-keys
        languageJavascript: `#e8696b`,
        javascript: `#e8696b`,
        background: `#292c34`,
        comment: `#5e6a76`,
        string: `#a8e2a8`,
        var: `#b3bac5`,
        number: `#e4854d`,
        constant: `#b3bac5`,
        plain: `#fff`,
        doctype: `#e8696b`,
        tag: `#e8696b`,
        keyword: `#d49fd4`,
        boolean: `#ff5874`,
        function: `#5F8DC3`,
        parameter: `#F9965D`,
        className: `#ffcf74`,
        attrName: `#bf87ba`,
        attrValue: `#a8e2a8`,
        interpolation: `#fff`,
        punctuation: `#5FA8AA`,
        ['maybe-class-name']: `#fff`,
        property: `#80cbc4`,
        propertyAccess: `#fff`,
        namespace: `#b2ccd6`,
        highlight: `rgba(255,255,255,0.07)`,
        highlightBorder: '#196FD8',
        dom: `#5F8DC3`,
        operator: `#5FA8AA`,
      },
      white: '#FFFFFF',
    },
    fontColor: 'var(--maximeheckel-colors-typeface-0, #2B2D3E)',
    foregroundColor:
      'var(--maximeheckel-colors-body-1, rgba(217, 230, 247, 0.55))',
    overlayBackground:
      'var(--maximeheckel-colors-body-2, rgba(236, 236, 236, 0.8))',
    transitionTime: 0.5,
  },

  dark: {
    backgroundColor: 'var(--maximeheckel-colors-body-0, #1b1e21)',
    borderColor: 'rgba(255, 255, 255, 0.1)',
    boxShadow: 'rgba(0,0,0,0.2) 0px 20px 40px',
    colors: {
      black: '#000000',
      blue: 'var(--maximeheckel-colors-brand, #5184f9)',
      gray: '#F8F8F9',
      pink: '#FED5D7',
      prism: {
        token: `#fff`,
        // tslint:disable-next-line:object-literal-sort-keys
        languageJavascript: `#e8696b`,
        javascript: `#e8696b`,
        background: `#0F111B`,
        comment: `#5e6a76`,
        string: `#a8e2a8`,
        var: `#b3bac5`,
        number: `#e4854d`,
        constant: `#b3bac5`,
        plain: `#fff`,
        doctype: `#e8696b`,
        tag: `#e8696b`,
        keyword: `#d49fd4`,
        boolean: `#ff5874`,
        function: `#5F8DC3`,
        parameter: `#F9965D`,
        className: `#ffcf74`,
        attrName: `#bf87ba`,
        attrValue: `#a8e2a8`,
        interpolation: `#fff`,
        punctuation: `#5FA8AA`,
        ['maybe-class-name']: `#fff`,
        property: `#80cbc4`,
        propertyAccess: `#fff`,
        namespace: `#b2ccd6`,
        highlight: `rgba(255,255,255,0.07)`,
        highlightBorder: '#3c83da',
        dom: `#5F8DC3`,
        operator: `#5FA8AA`,
      },
      white: '#FFFFFF',
    },
    fontColor: 'var(--maximeheckel-colors-typeface-0, #FFFFFF)',
    foregroundColor:
      'var(--maximeheckel-colors-body-1, rgba(14, 20, 27, 0.55))',
    overlayBackground: 'var(--maximeheckel-colors-body-2, rgba(0,0,0,0.40))',
    transitionTime: 0.5,
  },
};

export default theme;
