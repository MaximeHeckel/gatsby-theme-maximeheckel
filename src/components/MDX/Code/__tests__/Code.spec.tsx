/* eslint-disable react/no-unescaped-entities */
import { ThemeProvider } from 'emotion-theming';
import { cleanup, render } from '@testing-library/react';
import React from 'react';
import { Code, calculateLinesToHighlight, hasTitle, preToCodeBlock } from '../';
import theme from '../../../../theme';

afterEach(() => {
  cleanup();
});

describe('Code', () => {
  it('hasTitle returns the title part of a given metastring if present', () => {
    expect(hasTitle('hello,world,title=Test123')).toBe('Test123');
    expect(hasTitle('hello,world')).toBe('');
  });

  it('calculateLinesToHighlight returns the proper set of lignes to highlight of a given metastring', () => {
    expect(
      calculateLinesToHighlight(
        'javascript {7-12} title=Writing the rule (step 1)'
      )(1)
    ).toBeFalsy();
    expect(
      calculateLinesToHighlight(
        'javascript {7-12} title=Writing the rule (step 1)'
      )(7)
    ).toBeTruthy();

    expect(
      calculateLinesToHighlight('javascript title=Writing the rule (step 1)')(7)
    ).toBeFalsy();
  });

  it('preToCodeBlock returns the proper set of props for our code block components if the children are of type <code>', () => {
    const preProps = {
      children: {
        props: {
          children: 'some code to render',
          mdxType: 'code',
          metastring: 'javascript',
        },
      },
    };
    expect(preToCodeBlock(preProps)).toMatchSnapshot();

    const prePropsWithNoCode = {
      children: {
        props: {
          children: 'some inline code to render',
          metastring: 'javascript',
        },
      },
    };

    expect(preToCodeBlock(prePropsWithNoCode)).toBeUndefined();
  });

  it('Renders a Codeblock component when the proper preProps are passed', () => {
    const { container } = render(
      <ThemeProvider theme={theme.light}>
        <Code>
          <div metastring="javascript" mdxType="code">
            var hello="world"
          </div>
        </Code>
      </ThemeProvider>
    );

    expect(container.querySelector('pre[class="prism-code"]')).toBeDefined();
    expect(container.querySelector('span[class="number-line"]')).toBeDefined();
    expect(
      container.querySelector('span[class="number-line"]')
    ).toHaveTextContent(1);
  });

  it('Renders a Codeblock with title when the proper preProps are passed', () => {
    const { container } = render(
      <ThemeProvider theme={theme.light}>
        <Code>
          <div metastring="javascript title=test" mdxType="code">
            some code to render
          </div>
        </Code>
      </ThemeProvider>
    );

    expect(
      container.querySelector('p[data-testid="codesnippet-title"]')
    ).toHaveTextContent('test');
    expect(container.querySelector('pre[class="prism-code"]')).toBeDefined();
    expect(container.querySelector('span[class="number-line"]')).toBeDefined();
    expect(
      container.querySelector('span[class="number-line"]')
    ).toHaveTextContent(1);
    expect(container.querySelector('button')).toHaveTextContent('Copy');
  });

  it('Renders a Codeblock with title and line highlight when the proper preProps are passed', () => {
    const { container } = render(
      <ThemeProvider theme={theme.light}>
        <Code>
          <div metastring="javascript {1-3}title=test" mdxType="code">
            {`some code to render
            some code to render 2
            some code to render 3
            some code to render 4`}
          </div>
        </Code>
      </ThemeProvider>
    );

    expect(
      container.querySelector('p[data-testid="codesnippet-title"]')
    ).toHaveTextContent('test');
    expect(container.querySelector('pre[class="prism-code"]')).toBeDefined();
    expect(container.querySelector('span[class="number-line"]')).toBeDefined();
    expect(
      container.querySelectorAll('span[class="number-line"]')
    ).toHaveLength(4);
    expect(
      container.querySelectorAll('div[class="token-line highlight-line"]')
    ).toHaveLength(3);
    expect(container.querySelectorAll('div[class="token-line"]')).toHaveLength(
      1
    );
    expect(container.querySelector('button')).toHaveTextContent('Copy');
  });

  it('Renders a <pre> when there are no proper preProps passed', () => {
    const { container } = render(
      <ThemeProvider theme={theme.light}>
        <Code>
          <div metastring="javascript">var hello="world"</div>
        </Code>
      </ThemeProvider>
    );

    expect(container.querySelector('pre')).toBeDefined();
    expect(container.querySelector('pre[class="prism-code"]')).toBeNull();
    expect(container.querySelector('pre').firstElementChild).toHaveTextContent(
      'var hello="world"'
    );
  });
});
