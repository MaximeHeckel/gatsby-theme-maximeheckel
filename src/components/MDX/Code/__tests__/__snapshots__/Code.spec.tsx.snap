// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code preToCodeBlock returns the proper set of props for our code block components if the children are of type <code> 1`] = `
Object {
  "className": "",
  "codeString": "some code to render",
  "language": "",
  "mdxType": "code",
  "metastring": "javascript",
}
`;
